### PR TITLE
Deprecate net_pkt based CoAP library

### PR DESCRIPTION
**Pull request by rveerama1**
Socket based CoAP library is merged. So deprecating net_pkt based CoAP library.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: subsys/net/lib/coap/Kconfig [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #
 
 config COAP
-	bool "CoAP Support"
+	bool "CoAP Support[Deprecated]"
```
Comment (nashif) [inline subsys/net/lib/coap/Kconfig]: this is misleading, add more info that only net_pkt mode is being deprecated. I would also have done things differently, keep COAP kconfig and introduce COAP_NET_PKT and COAP_SOCK and avoid duplicating all of the Kconfigs

**Review by nashif — CHANGES_REQUESTED** [review]

File: subsys/net/lib/coap/Kconfig (hunk 1 above)
Comment (rveerama1) [reply subsys/net/lib/coap/Kconfig]: Added a bit more information. I didn't want to modify existing CoAP related stuff. Users of current CoAP can continue as it is except deprecated warnings.

**Review by pfalcon — APPROVED** [review]
Seems reasonable to me.

@nashif, please re-review.

@mike-scott , please review.

File: subsys/net/lib/coap/Kconfig (hunk 1 above)
Comment (rveerama1) [reply subsys/net/lib/coap/Kconfig]: > nashif@50bfcb8

Only reason for not doing like that was, you have changed everything in upstream Zephyr who are depending on CoAP. It fails to compile immediately for non upstream. This is the only reason I opted for separate folders and config. I am not arguing about that. When someone has to touch immediately their code after your proposal means why not keep only one implementation in one single go. For reviewers also it is bit easy to see only socket related changes.

**Comment by rveerama1** [comment]
Applied @nashif patch[https://github.com/nashif/zephyr/commit/50bfcb8] and deprecate patch.

**Comment by rveerama1** [comment]
@nashif ping

**Review by mike-scott — APPROVED** [review]
I'm ok w/ the depreciation as long as we keep the net_pkt support till the next Zephyr version.  My work has been extremely busy lately and I need more time w/ the socket conversion.

**Comment by rveerama1** [comment]
@nashif ping

**Review by nashif — APPROVED** [review]

**Pull request merged**